framework/src/application/node/forger/strategies.js

```diff
@@ -0,0 +1,118 @@
+		// components
+		logger,
+		// Modules
+		chainModule,
+		transactionPoolModule,
+		// constants
+		maxPayloadLength,
+	}) {
+		this.chainModule = chainModule;
+		this.transactionPoolModule = transactionPoolModule;
+		this.logger = logger;
+
+		this.constants = { maxPayloadLength: BigInt(maxPayloadLength) };
+	}
+
+	async getTransactionsForBlock() {
+		// Initialize array to select transactions
+		const readyTransactions = [];
+
+		// Initialize block size with 0
+		let blockPayloadSize = BigInt(0);
+
+		// Initialize state store which will be discarded after selection
+		const stateStore = this.chainModule.newStateStore();
+
+		// Get processable transactions from transaction pool
+		const processableTransactions = this.transactionPoolModule.getProcessableTransactions();
+
+		// Convert sender transactions to min heap by nonce
+		const transactionsBySender = Object.keys(processableTransactions).reduce(
+			(obj, senderId) => {
+				const minHeapByNonce = new MinHeap();
+				processableTransactions[senderId].forEach(t =>
+					minHeapByNonce.push(t.nonce, t),
+				);
+				obj[senderId] = minHeapByNonce;
+
+				return obj;
+			},
+			{},
+		);
+
+		// Loop till we have last account exhausted to pick transactions
+		while (Object.keys(transactionsBySender).length !== 0) {
+			const feePriorityHeap = new MaxHeap();
+
+			// Prepare max heap for high fee priority for lowest nonce of all available accounts
+			for (const senderId of Object.keys(transactionsBySender)) {
+				const lowestNonceTrx = transactionsBySender[senderId].peek().value;
+				feePriorityHeap.push(lowestNonceTrx.fee, lowestNonceTrx);
+			}
+
+			// Get the transaction with highest fee and lowest nonce
+			const lowestNonceHighestFeeTrx = feePriorityHeap.pop().value;
+
+			// Try to process transaction
+			const result = await this.chainModule.processTransactionsWithStateStore(
+				[lowestNonceHighestFeeTrx],
+				stateStore,
+			);
+
+			// If transaction can't be processed then discard all transactions
+			// from that account as other transactions will be higher nonce
+			if (result.transactionsResponses[0].status !== TransactionStatus.OK) {
+				delete transactionsBySender[lowestNonceHighestFeeTrx.senderId];
+
+				// eslint-disable-next-line no-continue
+				continue;
+			}
+
+			// If transaction byte size can't fit in max payload length
+			// then discard all transactions from that account as
+			// other transactions will be higher nonce
+			const trsByteSize = BigInt(lowestNonceHighestFeeTrx.getBytes().length);
+			if (blockPayloadSize + trsByteSize > this.constants.maxPayloadLength) {
+				delete transactionsBySender[lowestNonceHighestFeeTrx.senderId];
+				// eslint-disable-next-line no-continue
+				continue;
+			}
+
+			// Select transaction, increase block size and remove that transaction from heap
+			readyTransactions.push(lowestNonceHighestFeeTrx);
+			blockPayloadSize += trsByteSize;
+			transactionsBySender[lowestNonceHighestFeeTrx.senderId].pop();
```

Comment:
I think if you still have more transaction from the same sender, you want to put that in the heap

Reply:
That we do in start of the loop.